libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2111,57 +2111,61 @@ void TypeChecker::typeCheckABIEncodeCallFunction(FunctionCall const& _functionCa
 		return;
 	}
 
-	auto const functionPointerType = dynamic_cast<FunctionTypePointer>(type(*arguments.front()));
-
-	if (!functionPointerType)
+	FunctionType const* externalFunctionType = nullptr;
 	{
-		m_errorReporter.typeError(
-			5511_error,
-			arguments.front()->location(),
-			"Expected first argument to be a function pointer, not \"" +
-			type(*arguments.front())->canonicalName() +
-			"\"."
-		);
-		return;
+		auto const functionPointerType = dynamic_cast<FunctionTypePointer>(type(*arguments.front()));
+		if (!functionPointerType)
```

Comment:
Another way to avoid having this extra variable visible in the whole function, without introducing an artificial block, would be to do this:
```suggestion
		if (auto const functionPointerType = dynamic_cast<FunctionTypePointer>(type(*arguments.front())))
```